MIP/mip-58/README.md

```diff
@@ -0,0 +1,208 @@
+
+   ![Transaction History](tx-history.png)
+   Here users would be able to see if the bridge has been completed. It's either pending or completed.
+
+2. **Batch Completion**:
+   - Multisig relayers process pending transactions in batches during downtime, ensuring timely resolution.
+
+3. **Contract Simplification**:
+   - Combine lock and completion functionality on the counterparty contract.
+   - Remove refund logic to streamline operations and improve security.
+   - Cheaper transactions because of reduction of logic.
+   - Consolidate Initiator and Counterparty into a single contract (this might be the most dangerous thing proposed but it has already been proposed for current implementation).
+
+4. **Rate Limiting**:
+   - Rate limiting should be implemented on the L1 and maps each day to a budget, for each direction. Once the budget is reached on one of the directions, no more tokens can be transferred on that direction. The bridge is financially secured by an insurance fund maintained by Movement Labs and the maximum amount of tokens to be transferred per day, per direction is one quarter of the insurance fund balance. This is meant to account for the insurance fund to be able to insure all current funds already transferred and all tokens inflight, per direction.
+
+5. **Bridge Fee**:
+   - When bridging from L1 to L2, the protocol, through the Relayer, sponsors the gas cost on Movement. We do not need to make any modification on contracts or Relayer to support it.
+   - When bridging from L2 to L1, we have a few viable solutions but it's preferrable to highlight two.
+      1. Relayer sets a fee on L2, a global variable that can be set at any given time. Considering that block validation time on L1 is bigger than on L2, it becomes a viable approach since L2 can rapidly adjust the fee according to the current block and always charge an above gas cost fee to assure that despite hiccups, the bridge is net positive. $MOVE is deducted from the amount of tokens that are currently being bridged and transferred to a funds manager. Input of the user is on the final value of tokens that it should receive on the L1. This gives the protocol a very reliable way to estimate how much MOVE will be charged and feed to the user a precise amount of tokens.
+      2. Enable the Relayer to specify on the L1 completeBridgeTransfer transaction, the bridge fee per transaction. The amount is deducted from the total amount of tokens that were bridged and transferred to a funds manager. The dangerous situation is that we expect to way for over 10 minutes before the transfer can occur, and this could lead to a big disparity between the expected amout of funds and the actual amount of tokens received.
+
+
+[Solidity Implementation](https://github.com/movementlabsxyz/movement/tree/primata/simple-native-bridge)
```

Comment:
I think the code samples should be restored. In particular the code for initiate and complete, in both Move and Solidity, in particular showing how the parameter check works.

Reply:
but this is the code itself, isn't it?